checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -888,5 +888,75 @@ protected TypeMirror lubTypeFrame(TypeMirror a, TypeMirror b) {
 
             return TypesUtils.leastUpperBound(a, b, processingEnv);
         }
+
+        /**
+         * Compute the greatest lower bound of two initialization qualifiers. Returns null if one of
+         * the qualifiers is not in the initialization hierarachy. Subclasses should override
+         * greatestLowerBound and call this method for initialization qualifiers.
+         *
+         * @param anno1 an initialization qualifier
+         * @param qual1 a qualifier kind
+         * @param anno2 an initialization qualifier
+         * @param qual2 a qualifier kind
+         * @return the glb of anno1 and anno2
+         */
+        protected AnnotationMirror greatestLowerBoundInitialization(
+                AnnotationMirror anno1,
+                QualifierKind qual1,
+                AnnotationMirror anno2,
+                QualifierKind qual2) {
+            if (!isInitializationAnnotation(anno1) || !isInitializationAnnotation(anno2)) {
+                return null;
+            }
+
+            // Handle the case where one is a subtype of the other.
+            if (isSubtypeInitialization(anno1, qual1, anno2, qual2)) {
+                return anno1;
+            } else if (isSubtypeInitialization(anno2, qual2, anno1, qual1)) {
+                return anno2;
+            }
+            boolean unknowninit1 = isUnknownInitialization(anno1);
+            boolean unknowninit2 = isUnknownInitialization(anno2);
+            boolean underinit1 = isUnderInitialization(anno1);
+            boolean underinit2 = isUnderInitialization(anno2);
+
+            // Handle @Initialized.
+            if (isInitialized(anno1)) {
+                assert underinit2;
+                return FBCBOTTOM;
+            } else if (isInitialized(anno2)) {
+                assert underinit1;
+                return FBCBOTTOM;
+            }
+
+            if (underinit1 && underinit2) {
+                return createUnderInitializationAnnotation(
+                        glbTypeFrame(
+                                getTypeFrameFromAnnotation(anno1),
+                                getTypeFrameFromAnnotation(anno2)));
+            }
+
+            assert (unknowninit1 || underinit1) && (unknowninit2 || underinit2);
+            return createUnderInitializationAnnotation(
+                    glbTypeFrame(
+                            getTypeFrameFromAnnotation(anno1), getTypeFrameFromAnnotation(anno2)));
+        }
+
+        /**
+         * Returns the greatest lower bound of two types.
+         *
+         * @param a the first argument
+         * @param b the second argument
+         * @return the glb of the two arguments
+         */
+        protected TypeMirror glbTypeFrame(TypeMirror a, TypeMirror b) {
+            if (types.isSubtype(a, b)) {
```

Comment:
This is duplicated in `TypesUtils.greatestLowerBound`, so you could just call it directly.

Reply:
I don't see this special case in [`TypesUtils.greatestLowerBound`](https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java#L751).  Am I missing something?

Reply:
`TypesUtils.greatestLowerBound` calls `com.sun.tools.javac.code.Types#glb(com.sun.tools.javac.code.Type, com.sun.tools.javac.code.Type)` which has the special case.